motionplan/collision.go

```diff
@@ -311,3 +313,134 @@ func createUniqueCollisionMap(geoms []spatial.Geometry) (map[string]spatial.Geom
 	}
 	return geomMap, nil
 }
+
+// CheckPlan checks if obstacles intersect the trajectory of the frame following the plan.
+func CheckPlan(
+	frame referenceframe.Frame,
+	plan []map[string][]referenceframe.Input,
+	obstacles []*referenceframe.GeometriesInFrame,
+	fs referenceframe.FrameSystem,
+) (bool, error) {
+	// ensure that we can actually perform the check
+	if len(plan) == 0 {
+		return false, errors.New("cannot check validity of an empty plan")
+	}
+
+	// check if we are working with ptgs
+	ptgProv, ok := frame.(tpspace.PTGProvider)
+	if ok {
+		return checkPtgPlan(frame, plan, obstacles, fs, ptgProv.PTGs())
+	}
+
+	if len(plan) < 2 {
+		return false, errors.New("plan must have at least two elements")
+	}
+
+	// construct constraints
+	sf, err := newSolverFrame(fs, frame.Name(), referenceframe.World, plan[0])
+	if err != nil {
+		return false, err
+	}
+	worldState, err := referenceframe.NewWorldState(obstacles, nil)
+	if err != nil {
+		return false, err
+	}
+	collisionConstraints, err := createAllCollisionConstraints(sf, fs, worldState, referenceframe.StartPositions(fs), nil)
```

Comment:
> have the functionality of CheckPlan be to instantiate a planager

the `newPlanManager` function requires us to provide a `solverFrame` as one of its arguments, so I disagree that we could then get the solverFrame off of the planager instead of constructing it by hand.

> we're basically replicating `planManager.plannerSetupFromMoveRequest`

In order to use `planManager.plannerSetupFromMoveRequest` we would need to know what the goal is as a `spatialmath.Pose`. While is it easy to get this information from a non-PTG plan, I cannot think of a simple way to get that same information from a PTG-plan.
Perhaps, `goal referenceframe.PoseInFrame` can be appended to the function signature.. but that would require some major renovations on MoveOnGlobe since that is where we would be calling the function from and destinations are initially specified as a `geo.Point`...

let me know your thoughts

Reply:
You're correct regarding the solver frame; sorry about that.

> we would need to know what the goal is as a spatialmath.Pose. While is it easy to get this information from a non-PTG plan, I cannot think of a simple way to get that same information from a PTG-plan.

Look here: https://github.com/viamrobotics/rdk/pull/2792/files#r1303257056

Reply:
I will rebase on top of your changes

Reply:
rebased, but I am unsure how I can use rectifyTPspace to merge the two plan checking functions together